zipkin-storage/elasticsearch/src/main/java/zipkin2/elasticsearch/ElasticsearchStorage.java

```diff
@@ -265,37 +266,71 @@ void clear(String index) throws IOException {
     try {
-      HttpCall.Factory http = http();
-      AggregatedHttpRequest request = AggregatedHttpRequest.of(
-        HttpMethod.GET, "/_cluster/health/" + index);
-      return http.newCall(request, READ_STATUS, "get-cluster-health").execute();
+      ensureIndexTemplates(); // called only once, so we have to double-check health
+      AggregatedHttpRequest request = AggregatedHttpRequest.of(GET, "/_cluster/health/" + index);
+      CheckResult result = http().newCall(request, READ_STATUS, "get-cluster-health").execute();
+      if (result == null) throw new IllegalArgumentException("No content reading cluster health");
+      return result;
     } catch (Throwable e) {
       Call.propagateIfFatal(e);
-      // Unwrap the marker exception as the health check is not relevant for the throttle component,
-      // and wrapping interferes with humans intended to read this message.
-      return CheckResult.failed(e instanceof RejectedExecutionException ? e.getCause() : e);
+      // Wrapping interferes with humans intended to read this message:
+      //
+      // Unwrap the marker exception as the health check is not relevant for the throttle component.
+      // Unwrap any IOException from the first call to ensureIndexTemplates()
+      if (e instanceof RejectedExecutionException || e instanceof UncheckedIOException) {
+        e = e.getCause();
+      }
+      return CheckResult.failed(e);
     }
   }
 
-  @Memoized // since we don't want overlapping calls to apply the index templates
+  volatile IndexTemplates indexTemplates; // visible for testing
+
+  // Memoized since we don't want overlapping calls to apply the index templates
   IndexTemplates ensureIndexTemplates() {
+    if (indexTemplates == null) {
+      synchronized (this) {
+        if (indexTemplates == null) indexTemplates = doEnsureIndexTemplates();
+      }
+    }
+    return indexTemplates;
+  }
+
+  IndexTemplates doEnsureIndexTemplates() {
     try {
-      IndexTemplates templates = new VersionSpecificTemplates(this).get();
       HttpCall.Factory http = http();
-      ensureIndexTemplate(http, buildUrl(templates, SPAN), templates.span());
-      ensureIndexTemplate(http, buildUrl(templates, DEPENDENCY), templates.dependency());
-      ensureIndexTemplate(http, buildUrl(templates, AUTOCOMPLETE), templates.autocomplete());
+      IndexTemplates templates = versionSpecificTemplates(http);
+      ensureIndexTemplate(http, buildUrl(templates, TYPE_SPAN), templates.span());
+      ensureIndexTemplate(http, buildUrl(templates, TYPE_DEPENDENCY), templates.dependency());
+      ensureIndexTemplate(http, buildUrl(templates, TYPE_AUTOCOMPLETE), templates.autocomplete());
       return templates;
     } catch (IOException e) {
-      throw Platform.get().uncheckedIOException(e);
+      throw new UncheckedIOException(e);
     }
   }
 
+  IndexTemplates versionSpecificTemplates(HttpCall.Factory http) throws IOException {
+    Float version = ElasticsearchVersion.INSTANCE.get(http);
```

Comment:
fixed this in the ElasticsearchVersion type

```suggestion
    float version = ElasticsearchVersion.INSTANCE.get(http);
```